src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4408,18 +4408,57 @@ void setAnalyzedNullabilityAsContinuation(
                 };
             }
 
-            static Symbol? getTargetMember(TypeSymbol containingType, BoundObjectInitializerMember objectInitializer)
+            Symbol? getTargetMember(TypeSymbol containingType, BoundObjectInitializerMember objectInitializer)
             {
                 var symbol = objectInitializer.MemberSymbol;
+                if (symbol == null)
+                    return null;
 
-                // https://github.com/dotnet/roslyn/issues/78828: adjust extension member based on new containing type
-                if (symbol != null && !symbol.GetIsNewExtensionMember())
+                if (!symbol.GetIsNewExtensionMember())
                 {
                     Debug.Assert(TypeSymbol.Equals(objectInitializer.Type, GetTypeOrReturnType(symbol), TypeCompareKind.IgnoreNullableModifiersForReferenceTypes));
-                    symbol = AsMemberOfType(containingType, symbol);
+                    return AsMemberOfType(containingType, symbol);
                 }
 
-                return symbol;
+                if (symbol.OriginalDefinition.ContainingType is not { Arity: not 0 } extension)
```

Comment:
>not { Arity: not 0 } [](http://example.com/codeflow?start=64&length=20)

I think this double negative is hard to follow. Consider converting to a positive check. We don't need to worry about a `null` case due to ```GetIsNewExtensionMember``` check above.